content/doc/reference/reference-environment-variables.md

```diff
@@ -174,6 +174,9 @@ If `TAILSCALE_LOGIN_SERVER` is provided, the agent will be configured to reach a
 |  Name  |  Description  |
 |-----------------------|------------------------------|
 |[`CC_HASKELL_STACK_TARGET`](/doc/applications/haskell/#specify-stack-package-target "Specify Stack package target") | Specify Stack package target. |
+|`CC_HASKELL_STACK_SETUP_COMMAND` | Only use this variable to override the default `SETUP` step when running Haskell |
+|`CC_HASKELL_STACK_INSTALL_COMMAND` | Only use this variable to override the default `INSTALL` step when running Haskell |
+|`CC_HASKELL_STACK_INSTALL_DEPENDENCIES_COMMAND` | Only use this variable to override the `INSTALL DEPENDENCIES` one when running Haskell |
```

Comment:
My 2cts ont this: 
- Caps are not necessary to the "step" names (on the right)
- We should precise these are related to stack deployment, not Haskell

For example: 
```suggestion
|`CC_HASKELL_STACK_SETUP_COMMAND` | Only use this variable to override the default `setup` Stack step command |
|`CC_HASKELL_STACK_INSTALL_COMMAND` | Only use this variable to override the default `install` Stack step command |
|`CC_HASKELL_STACK_INSTALL_DEPENDENCIES_COMMAND` | Only use this variable to override the default `install --only-dependencies` Stack step command |
```

We could also add a link to `/doc/applications/haskell/#mandatory-configuration` in a way similar to `CC_HASKELL_STACK_TARGET`

BTW, I'm not sure the `CC_RUN_COMMAND` is necessary here, as it's a command used for every runtime (except Docker). But maybe this should be discussed in another issue/PR

Reply:
Thanks David :) 
I agree with the points you made for the "steps" part. 

Regarding the  `CC_RUN_COMMAND` i'd rather leave it. If we remove it for only Haskell but not the rest, it might seem strange for the user. 
Plus i feel it's better to remind user that they can use this :smile:

Reply:
Yep my proposal was more to remove it from all other runtimes and let it just at the "common" part at the beginning, it's why I suggest to discuss it in another issue/PR